### PR TITLE
Add bloom filter for single column equivalent expression

### PR DESCRIPTION
**Pull request by junli1026**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add bloom filter for single column equivalent expression like "name = 'batman'".

## Changelog

- New Feature

## Related Issues

Related #3832

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by junli1026** [comment]
Seems fasthash library faied aarch64 build

**Comment by PsiACE** [comment]
Perhaps we could implement the bloom filter with another hashing algorithm, such as seahash. Or later I can add a 64-bit CityHash v102 implementation to naive-cityhash.

**Comment by PsiACE** [comment]
I have just released naive-cityhash v0.2.0, supporting `cityhash64`, `cityhash64_with_seed` and `cityhash64_with_seeds`. The only problem is that `City64Hasher` and `City128Hasher` have not been implemented yet :(

**Comment by junli1026** [comment]
> I have just released naive-cityhash v0.2.0, supporting `cityhash64`, `cityhash64_with_seed` and `cityhash64_with_seeds`. The only problem is that `City64Hasher` and `City128Hasher` have not been implemented yet :(

It is fine, we can use SeaHash for now. Thanks for the following up.

**Comment by sundy-li** [comment]
Docs about the cityhash in ClickHouse:

https://go-faster.org/docs/clickhouse/hash

File: query/src/storages/index/bloom_filter.rs [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+}
+
+/// A bloom filter implementation for data column and values.
+///
+/// Most ideas/implementations are ported from Clickhouse.
+/// https://github.com/ClickHouse/ClickHouse/blob/1bf375e2b761db5b99b0f403b90c412a530f4d5c/src/Interpreters/BloomFilter.cpp
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct BloomFilter {
+    // Container for bitmap
+    container: Vec<u64>,
```
Comment (sundy-li) [inline query/src/storages/index/bloom_filter.rs:253]: Can we use `BitVec ` here?

File: common/datavalues/src/data_hasher.rs [hunk 2]
```diff
@@ -18,28 +18,33 @@ use std::hash::Hasher;
...
+use seahash::SeaHasher;
 
 /// TODO:
 /// This is very slow because it involves lots of copy to keep the origin state
-/// We should have our custom none-state hashe functions
-#[derive(Clone, Debug)]
+/// We should have our custom none-state hash functions
+/// Tracked work item: https://github.com/datafuselabs/databend/issues/3897
+#[derive(Clone)]
 pub enum DFHasher {
```
Comment (PsiACE) [inline common/datavalues/src/data_hasher.rs:28]: :+1:

**Review by sundy-li — APPROVED** [review]

**Review by PsiACE — APPROVED** [review]

**Pull request merged**